### PR TITLE
Fix issue#9, improve readme, and issue#5 make distributable

### PR DESCRIPTION
**Pull request by schalkneethling**
@alexgibson r?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,58 @@
...
+css/infobar.css
+js/infobar.js
+```
+
+### Using NPM
+
+You can install it as a NPM module
+
+``` bash
+npm install mozilla-infobar
```
Comment (alexgibson) [inline README.md]: Is this lib ready to be published on npm? Seeing as it is to be run in the browser, wouldn't this require support for consuming the library as a CommonJS module (or similar?)

File: README.md [hunk 2]
```diff
@@ -1,19 +1,58 @@
 # mozilla-infobar
 
+This is the home of the Mozilla, Firefox update, and translation bars. These bars, in the first case, provides an easy way to let your users know that there is a newer version of Firefox available, and that they should upgrade.
```
Comment (alexgibson) [inline README.md]: typo: `provide`

File: README.md [hunk 3]
```diff
@@ -22,36 +61,40 @@ Next include the base HTML of the InfoBar anywhere inside the HTML of your docum
...
-```
+In the first instance, you would add `link` elements to the `head` of your document. For example:
+
+``` html
 <link rel="alternate" hreflang="es-ES" href="http://www.mozilla.org/es-ES/firefox/new/" title="English (US)" />
 <link rel="alternate" hreflang="fr" href="http://www.mozilla.org/fr/firefox/new/" title="Français" />
 ```
 
-These list the alternate languages the current page is available in, as well as the URL to redirect the user to. The alternative is to make use of a language selector, using a `select` element.
+These list the alternate languages the current page is available in, as well as the URL to redirect the user to. For smaller website, or those only supporting a small selection of locales, hard coding these may be an option. Should your site support many locales, it would be wise to investigate generating these `link` elements using a server side language, coupled with template tags if required.
```
Comment (alexgibson) [inline README.md]: typo: `smaller websites`

File: README.md [hunk 4]
```diff
@@ -22,36 +61,40 @@ Next include the base HTML of the InfoBar anywhere inside the HTML of your docum
...
-If the value above is translate, it will show the Translation Bar based in the precense of some additional elements as described later. You can also specify both, as such:
+If the value above is translate, it will show the Translation Bar based on the precense of some additional elements as described later. You can also specify both, as such:
 
 ``` html
 <div ... data-infobar="update translate">
-....
 ```
 
-The order here is important, as wo InfoBar elements will never be shown simultaneously. In the scenario above, the first time the page is loaded, the logic for the update bar will be run. Once the user has taken the required action or, dismissed the notification, the translate bar's logic will b run on subsequent page loads.
+The order here is important, as both InfoBar elements will never be shown simultaneously. In the scenario above, the first time the page is loaded, the update bar will be run. Once the user has taken either of the available actions, the Translation Bar will be shown on subsequent page loads.
```
Comment (alexgibson) [inline README.md]: Suggest we stick to using `Update Bar` and `Translation Bar` using title case for consistency.

File: bower.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    "css/libs/",
+    "js/libs/",
+    "node_modules",
+    "tests",
+    ".gitignore",
+    ".jshintrc.dist",
+    "bower.json",
+    "contribute.json",
+    "LICENSE",
+    "package.json",
```
Comment (alexgibson) [inline bower.json]: Nit: remove trailing comma `,`

File: css/infobar.css [hunk 6]
```diff
@@ -44,3 +49,7 @@
 .mozilla-infobar [dir="rtl"] li {
     margin: 0 0 0 10px;
 }
+
+.hidden {
```
Comment (alexgibson) [inline css/infobar.css]: This is a very generic and common selector. Suggest name-spacing to avoid unintended breakage in other sites?

**Comment by alexgibson** [comment]
Looks good, main question is relating to npm publishing, but otherwise just some minor nits.

Not related to this PR specifically, but perhaps something that should be handled in another issue:
- Is there any technical reason why the unit tests can't/aren't run in Travis?
- It looks like we lost some of the unit test coverage when this came over from bedrock? We have the most common scenarios covered here it seems, but I remember Kohei patched quite a few edge case bugs that cropped up relating to the order of accept-language headers. Perhaps we could expand those a bit here too? The code is all in the bedrock test dir still: https://github.com/mozilla/bedrock/blob/master/tests/unit/spec/tabzilla.js

File: README.md (hunk 1 above)
Comment (schalkneethling) [inline README.md]: Yes, I reckon after this PR it will be ready to publish so, I added the docs. In terms of CommonJS, we could add that but, not sure if it is worth it. The main reason I added "support" for NPM here is because it was requested for Tabzilla which is essentially a CSS file with some assets.

**Comment by schalkneethling** [comment]
"Is there any technical reason why the unit tests can't/aren't run in Travis?"

Nope. This is a to-do

I will also look at getting additional test coverage in here. Thanks for the the review @alexgibson

File: README.md (hunk 1 above)
Comment (alexgibson) [inline README.md]: Hmm yeah, but a CSS file doesn't need anything special to be able to require-able? The standard for JS published on npm seems to be CommonJS support for things that can run in the browser (unless I'm behind the times :smile:). If this kind of support isn't planned, would it be worth adding some extra info to the docs explaining how people should use the library when using npm specifically?

Comment (alexgibson) [inline README.md]: Maybe something we could add that's applicable for both npm and bower? http://bower.io/#use-packages

Comment (schalkneethling) [inline README.md]: Yeah, let me take a look. Thanks @alexgibson

Comment (schalkneethling) [inline README.md]: I am wondering, perhaps we should only do this as a project clone or Bower and then see whether there are requests to make this available via NPM. Should there be people requesting that, we can find out whether it being "requireable" is a requirement (awkward sentence ;)) Thoughts @alexgibson

Comment (alexgibson) [inline README.md]: I agree if this is to be on npm then I think we should use UMD or similar pattern to support different loaders, else it will likely be a source of confusion for whomever tries to use it.

**Comment by schalkneethling** [comment]
@alexgibson updated.

File: package.json [hunk 7]
```diff
@@ -2,20 +2,22 @@
...
+    "test": "karma start tests/unit/karma.conf.js"
   },
   "author": "Mozilla",
   "license": "MPL2",
+  "dependencies": {},
   "devDependencies": {
     "karma": "^0.13.19",
     "jasmine": "^2.4.1",
-    "karma-firefox-launcher": "^0.1.6",
-    "karma-jasmine": "^0.3.6"
```
Comment (alexgibson) [inline package.json:19]: I think this is still needed to run the unit tests, right?

Comment (alexgibson) [inline package.json:19]: On a fresh install using this branch, running `npm test` gives: 

```
Error: No provider for "framework:jasmine"! (Resolving: framework:jasmine)
```

Comment (schalkneethling) [inline package.json:19]: Yup, merge fail. Adding back

File: package.json [hunk 8]
```diff
@@ -2,20 +2,22 @@
...
   },
+  "bugs": {
+      "url": "https://github.com/mozilla/mozilla-infobar/issues/new"
+  },
   "scripts": {
-    "test": "karma start tests/unit/karma.conf.js"  
+    "test": "karma start tests/unit/karma.conf.js"
   },
   "author": "Mozilla",
   "license": "MPL2",
```
Comment (alexgibson) [inline package.json]: Not related to this change, running `npm install` gives the following warning:

```
npm WARN package.json mozilla-infobar@0.2.0 license should be a valid SPDX license expression
```

Should this value be `MPL-2.0`?

**Comment by alexgibson** [comment]
Also see this npm warning when installing dev-dependencies:

```
npm WARN peerDependencies The peer dependency jasmine-core@* included from karma-jasmine will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency 
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
```

Probably worth adding `jasmine-core` as a direct dev-dependency here too?

**Comment by schalkneethling** [comment]
So, I just added jasmine-core as apposed to jasmine as a dependency, and I now do not see the warning. Can you confirm?

**Comment by alexgibson** [comment]
r+
